### ISSUE TITLE
.NET Core SignalR enforce authentication

### ISSUE THREAD

**Issue by kofifus** [aspnet/SignalR#2443]
Using .NET Core 2.1 & VS2017 preview 2 I created a simple web server with "Identity as UI" as explained [here](http://www.talkingdotnet.com/how-to-scaffold-identity-ui-in-asp-net-core-2-1/) and then added a SignalR chat following [this][1]. 

In particular I have:

```
	app.UseAuthentication();
	app.UseSignalR((options) => {
		options.MapHub<AgentHub>("/hubs/myhub");
	});
```

I now start the debugger which starts the browser and I make sure the user is _not_ logged in and go to http://localhost:5000/SignalRtest (my razor page that uses signalr.js), and since the user is not logged in I am correctly redirected to the login page. I then login and the chat works properly.

So far so good.

I now created a .NET Core 2.1 console application like this:

```
	class Program
	{
		public static async Task SetupSignalRHubAsync()
		{
			var hubConnection = new HubConnectionBuilder()
						.WithUrl("http://localhost:5000/hubs/myhub")
						.Build();

			await hubConnection.StartAsync();
			await hubConnection.SendAsync("Send", "consoleapp");
		}

		public static void Main(string[] args)
		{
			SetupSignalRHubAsync().Wait();
		}
	}
```

I run the console app and I see the  message on the browser client ! (I see "null joined" and "null: consoleapp")

So it seems the server allowed the console client (null) to connect with SignalR even through it did not authenticate.

- Why is this ? 
- How can I ensure no client can use SignalR without Authenticate ? 
- Once that is done how do I Authenticate from the SignalR console client ?

Thanks !

  [1]: https://docs.microsoft.com/en-us/aspnet/core/signalr/hubs?view=aspnetcore-2.1

**Comment by BrennanConroy** [comment]
Put `UseAuthentication` above `UseSignalR`, middleware runs in the order you place it.

**Comment by kofifus** [comment]
No that did not make any difference

**Comment by kofifus** [comment]
I realized I missed [Authorize]  for my hub.

Now the console client does not connect.

How do I authenticate the client with a username (email) and password ?

**Comment by kofifus** [comment]
ok will close this now and open another issue regarding authentication
